### PR TITLE
Create technical overview for new contributors

### PR DESCRIPTION
**Pull request by johnnyman727**
This overview should really help speed new contributors up as to the architecture of the system. It features relevant Github links, system diagrams, and a walk through of what happens when you push code.

### CODE REVIEW

File: T2-TECHNICAL-OVERVIEW.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+- [System Architecture](#system-architecture)
+    - [Hardware Overview](#high-level-look-at-hardware)
+    - [Software Overview](#software-architecture)
+- [Code Deploy Walkthrough](#a-code-deploy-example)
+- [Getting Questions Answered](#questions-)
+- [Links to more information](#more-useful-links)
+
+## Relevant Github Repositories
+
+###Tessel 2 Tools:
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: Confusing title

File: T2-TECHNICAL-OVERVIEW.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
+# Tessel 2 Technical Overview
+This guide is intended to help code contributors understand how relevant system components of Tessel 2 work, where to find code and design files, and accelerate the development process through technical transparency. 
+- [T2 Github Repositories](#relevant-github-repositories)
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: Make the ToC match all section headings

File: T2-TECHNICAL-OVERVIEW.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+###Tessel 2 Tools:
+You can find all of the Tessel repositories [on the organization page](https://github.com/tessel/) but this section provides some guidance over relevant Tessel 2-specific repositories.
+
+#####[Tessel 2 Command Line Interface](https://www.github.com/tessel/t2-cli) (CLI)
+
+The CLI is the primary method of interacting with Tessel for users and develoeprs. It provides useful functions like listing available Tessels, deploying code, and setting wifi credentials on the device.
+
+#####[Tessel 2 Virtual Machine](https://www.github.com/tessel/t2-vm) (VM)
+The VM is used primarily by Tessel developers who either don't have hardware available or want to develop on a faster computer. The repo provides the ability to create and run these Tessel virtual machines. You can use the CLI (see above) to interact with the running VM just as you would an actual Tessel 2. The VM has the limtation of not being able to set wifi credentials (it passes through the host network interface) or using hardware modules but USB devices do get passed through.
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: The last sentence is split infinitive. Break it up/shorten?

File: T2-TECHNICAL-OVERVIEW.md (hunk 1 above)
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: This split-out is a bit weird to me. I would look for CLI, firmware, OS all together as the core, then look for the hardware, then look for the tools

File: T2-TECHNICAL-OVERVIEW.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+#####[Tessel 2 Compiler](https://www.github.com/tessel/t2-compiler)
+The Tessel 2 compiler is another virtual machine with all the build tools needed to develope [native add-ons](https://nodejs.org/api/addons.html) to Node modules. These add-ons are built to perform faster or interact with lower level hardware than JS alone would provide. The compiler is being used, for example, to develop the [audio-video Node module](https://github.com/tessel/node-audiovideo) for webcams and recording audio from microphones.
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: typo "develope"

File: T2-TECHNICAL-OVERVIEW.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    
+### Device Related Code
+#####[Tessel 2 Operation System (OpenWRT)](https://www.github.com/tessel/openwrt-tessel)
+The primary processpr of the Tessel 2 runs a very lightweight version of Linux called OpenWRT. OpenWRT provides all of the TCP/IP drivers, threading/schedule support, and runs Node, Rust or whatever other language you're using with Tessel 2. This repo contains all of the source files of the Tessel build of OpenWRT.
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: typo "processpr"

File: T2-TECHNICAL-OVERVIEW.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+#####[Tessel 2 Co-processor Firmware](https://www.github.com/tessel/t2-firmware)
+Tessel 2 features a co-processor that manages a handful of responsibilities like routing USB communication from the CLI and the realtime operations on the GPIO pins of the two module ports. This repo contains not only the firmware that drives this microcontroller but hardware-related scripts that define the interface to other languages (like [tessel.js](https://www.github.com/tessel/t2-firmware/node/tessel.js) for Node scripts) so that the available hardware functionality doesn't get out of sync with what is exposed to the OpenWRT processor.
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: Would be good to call out the SAMD

File: T2-TECHNICAL-OVERVIEW.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+### Hardware Designs
+#####[Tessel 2 Hardware Design Files](https://www.github.com/tessel/t2-hardware)
+These are the actual schematic and assembly files for the Tessel 2 hardware. You can find information about the parts that are used and how they are all connected to each other. These files were created with [KiCad](http://www.kicad-pcb.org/display/KICAD/KiCad+EDA+Software+Suite), a completely open source electronics design tool.
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: "actual"?

File: T2-TECHNICAL-OVERVIEW.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+#####[Tessel Parts Library](https://github.com/tessel/tm-kicad-library)
+This repo contains all of the KiCad part models for all Tessel hardware (not just Tessel 2).
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: How do you use this? Do you need this in order to use the hardware design files? Or, these are the files that the schematics were generated from? Not sure what this repo is

File: T2-TECHNICAL-OVERVIEW.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+
+### Modules
+
+##### 10-Pin Modules
+There are 9 compatible 10-pin hardware modules for Tessel 2 available at the time of this writing:
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: Write a little more here. These repos give you what? Code? Hardware?

File: T2-TECHNICAL-OVERVIEW.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+* [Accelerometer Module](https://github.com/tessel/accel-attx4)
+* [Bluetooth Low Energy Module](https://github.com/tessel/ble-ble113a)
+* [Climate (temperature & humidity) Module](https://github.com/tessel/climate-si7020)
+* [GPS Module](https://github.com/tessel/gps-a2235h)
+* [Infrared Module](https://github.com/tessel/ir-attx4)
+* [Relay Module](https://github.com/tessel/relay-mono)
+* [Servo Module](https://github.com/tessel/servo-pca9685)
+
+##### USB Modules
+USB Modules haven't been released yet so some don't have Github repos and many are in progress. In general, Tessel supports the following kind of USB modules which you can find from many online vendors:
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: These are already supported? You mention that many are in progress; where can I see the in-progress ones?

File: T2-TECHNICAL-OVERVIEW.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+* Flash Storage Devices (no Github module, it just works through the Node `fs` module and similar for other languages)
+
+
+## System Architecture
+
+### High Level Look At Hardware
+Below is a high level diagram of the hardware architecture:
+![T2 Hardware Diagram](https://docs.google.com/drawings/d/124Qrry4MCywzKJq1mQSo8Xwk_S58YT4IG4OHf_2oUWM/pub?w=960&h=720)
+
+As you an see from the diagram, the Atmel co-processor manages the USB connection and can communicate with the MediaTek through a SPI bus. The MediaTek can execute user scripts, and when it interprets hardware commands (like pulling a GPIO high or sending data over I2C), it packages it up and sends it over to the Atmel over a pre-defined protocol on that SPI bus.
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: typo "As you an see" though I'd just cut that clause

File: T2-TECHNICAL-OVERVIEW.md [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+### Software Architecture
+The MediaTek is responsible for managing the two USB Host ports, the ethernet port, and the WiFi network interace directly from the OpenWRT Operating System.
+
+Let's take a look at the software architecture of Tessel 2:
+![T2 Software Architecture Diagram](https://docs.google.com/drawings/d/1PHXdbOQEIMa7lVsBzazaMPgGdrh_LRcIpF4FlbPXmxo/pub?w=782&h=1097)
+
+
+The USB Port on the Atmel actually has three interfaces to the to the MediaTek:
+1. A UART connection that acts a serial terminal to the MediaTek. You can use a program like `dterm` to access the console through this connection.
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: Make dterm a link?

File: T2-TECHNICAL-OVERVIEW.md [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+
+#### Why Domain Sockets?
+The beauty of this architecture with domain sockets is that most legitimate programming langauges have the capability to utilize domain sockets. This is how Tessel 2 can be used by any language, not just JavaScript (the code that packages up the commands into array byes just has to be ported to that language first).
```
Comment (Frijol) [inline T2-TECHNICAL-OVERVIEW.md]: Elaborate?

**Comment by johnnyman727** [comment]
@Frijol thanks for the feedback. I've updated the copy with all of your suggestions.

Instead of merging into the `projects` repo, I'm going to close this PR, create an `onboarding` repo, and add it there.

**Pull request closed without merging**